File: Go/Random/random-uuid.go
package main

import (
	"crypto/rand"
	"fmt"
	"log"
)

func main() {
	// Generate a random uuid.
	fmt.Println(uniqueUUIDGenerator())
}

// Generate a random uuid and return it as a string.
func uniqueUUIDGenerator() string {
	randomBytes := make([]byte, 16)
	_, err := rand.Read(randomBytes)
	if err != nil {
		log.Fatalln(err)
	}
	return fmt.Sprintf("%x-%x-%x-%x-%x", randomBytes[0:4], randomBytes[4:6], randomBytes[6:8], randomBytes[8:10], randomBytes[10:])
}

File: TypeScript/Random/random-uuid.ts
import * as crypto from "crypto"

function main(): void {
    // Generate a random UUID.
    console.log(uniqueUUIDGenerator())
}

main()

// Generate a random UUID and return it as a string.
function uniqueUUIDGenerator(): string {
    return randomStringGenerator(4) + "-" + randomStringGenerator(2) + "-" + randomStringGenerator(2) + "-" + randomStringGenerator(2) + "-" + randomStringGenerator(6)
}

// Generate a random string of the specified length.
function randomStringGenerator(length: number): string {
    return crypto.randomBytes(length).toString("hex")
}